src/Compatibility/ApiDiff/Microsoft.DotNet.ApiDiff.Tool/Microsoft.DotNet.ApiDiff.Tool.csproj

```diff
@@ -0,0 +1,22 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>$(NetToolMinimum)</TargetFramework>
+    <OutputType>Exe</OutputType>
+    <!-- <IsPackable>true</IsPackable>
+    <PackAsTool>true</PackAsTool>
+    <ToolCommandName>genapidiff</ToolCommandName> -->
+    <PackageDescription>Tool to emit markdown diffs between sets of assemblies.</PackageDescription>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <ProjectReference Include="..\..\GenAPI\Microsoft.DotNet.GenAPI\Microsoft.DotNet.GenAPI.csproj" />
```

Comment:
I was under the assumption that types were pushed down into ApiSymbolExtensions so that this new tool doesn't depend on GenAPI directly. Maybe I was wrong. Can you please tell how much code is used from that code base?

Would it make sense to create a new class library that is used by both GenAPI and ApiDiff so that this dependency is avoided?

Reply:
I think this isn't failing in source-build as in that build mode, the root sln doesn't get built. Instead a solution filter file gets used which doesn't have a reference to this new tool: https://github.com/dotnet/sdk/blob/d6bc7918c0be5005426fbbff43dfb622fc295142/eng/DotNetBuild.props#L14

Do we need to have this tool available when building source-only?

Reply:
No, this tool isn't needed for source build. It's meant for manual use.

Reply:
Cool. Please also respond to my first comment in this discussion about the layering of class libraries.

Reply:
> Can you please tell how much code is used from that code base?

It's basically the code that I had to refactor in my previous PRs. You can refer to those.

> Would it make sense to create a new class library that is used by both GenAPI and ApiDiff so that this dependency is avoided?

I'd like to understand this better. I'm not sure how this would be a problem, since ApiDiff won't be used in source-build, but takes a dependency on GenAPI.